TMDBApp/TMDBApp/ListMovieViewController.swift

```diff
@@ -0,0 +1,177 @@
+    
+    // collections
+    var movies = [Movie]()
+    var genres = [Genrer]()
+    
+    // control pages
+    var currentPage = 1
+    var totalPages = 0
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.updateUI()
+        self.getGenres()
+    }
+    
+    fileprivate func updateUI(){
+        
+        self.view.backgroundColor = UIColor.init(hexString: Colors.defaultDarkBlue.rawValue)
+        self.tableView.backgroundColor = UIColor.init(hexString: Colors.defaultDarkBlue.rawValue)
+        
+        if query.isEmpty{
+            self.title = "Upcoming Movies"
+        }else{
+            self.title = query
+        }
+    }
+
+    fileprivate func getGenres(){
+        
+        self.showProgress()
+        
+        Requests.sharedInstance().requestGenrer{ [weak self] (result, error) in
+            
+            guard result != nil, let resultDict = result as? DictionaryData else{
+                self?.showError(error.debugDescription)
+                return
+            }
+            self?.parseGenrer(resultDict)
+            
+            self?.getMovies()
+        }
+    }
+    
+    fileprivate func getMovies(){
+        
+        self.showProgress()
+        
+        Requests.sharedInstance().requestMovies(fromPage: currentPage, withQuery: query) { [weak self] (result, error) in
+            
+            guard result != nil, let resultDict = result as? DictionaryData else{
+                self?.showError(error.debugDescription)
+                return
+            }
+            self?.parseMovie(resultDict)
+            
+            self?.tableView?.reloadData()
+            
+            self?.dismissProgress()
+        }
+    }
+    
+    fileprivate func parseGenrer(_ data:DictionaryData){
+        guard let results = data[jsonResult.genres.rawValue] as? [DictionaryData] else {
+            return
+        }
+        for result in results{
+            let genrer = Genrer(withDict: result)
+            genres.append(genrer)
+        }
+    }
+    
+    fileprivate func parseMovie(_ data:DictionaryData){
+        guard let results = data[jsonResult.results.rawValue]    as? [DictionaryData],
+              let total   = data[jsonResult.totalPages.rawValue] as? Int else {
+            return
+        }
+        totalPages = total
+        
+        for result in results{
+            let movie = Movie(withDict: result, andGenres: self.genres)
+            movies.append(movie)
+        }
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    func showError(_ error: String){
+        print(error)
+        updatesOnMain{
+            self.dismissProgress()
+            
+            let alert = UIAlertController(title: "Error!", message: "Check your internet connection. An error has occurred on getting data \(error)", preferredStyle: .alert)
```

Comment:
You presented the search result options in various languages. How would be the right procedure for the whole app to be internationalized instead of the search results only?